src/utils.php

```diff
@@ -36,35 +36,56 @@ function create_list_of_cars(array $cars): string | false
     return $ret;
 }
 
-function validateDataFields(string $name, int $year_made, float $zero_sixty, float $price, string $brand): string
+function validateDataFields(Car $car): bool
 {
-    $err_msg = "";
-    if (!is_string($name) || empty($name) || strlen($name) > 30 || strlen($name) <= 0) {
-        $err_msg .= "Invalid name" . "<br>";
+    $ret = true;
+
+    if (
+        empty($car->name) == true ||
+        strlen($car->name) > 30 ||
+        strlen($car->name) <= 0
+    ) {
+        echo "Name cannot be over 30 characters long" . "<br>";
+        $ret = false;
     }
 
-
-    if (empty($year_made) || !is_numeric($year_made) || strlen(strval($year_made)) <= 0 || strlen(strval($year_made)) > 4 || $year_made < 0 || $year_made > 2023) {
-        $err_msg .= "Invalid year" . "<br>";
+    if (
+        empty($car->year_made) == true ||
+        is_int($car->year_made) == false ||
+        strlen(strval($car->year_made)) <= 0 ||
+        strlen(strval($car->year_made)) > 4 ||
+        $car->year_made < 0 ||
+        $car->year_made > 2023
+    ) {
+        echo "Invalid year" . "<br>";
```

Comment:
Dont forget you shouldn't be echoing from within a function, it makes it hard to test